docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,502 @@
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kotlin.compiler.execution.strategy` system property used to choose a compiler execution strategy. 
+> `use the Gradle property `kotlin.compiler.execution.strategy` or the compile task property `compilerExecutionStrategy` instead
+>
+> **Deprecation cycle:**
+>
+> - 1.7.0: raise the deprecation level to a warning
+> - \> 1.7.0: remove the property
+
+### Remove kotlinOptions.jdkHome compiler option
+
+> **Issue**: [KT-46541](https://youtrack.jetbrains.com/issue/KT-46541)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlinOptions.jdkHome` compiler option used to include a custom JDK from the specified location 
+> into the classpath instead of the default `JAVA_HOME`. Use [Java toolchains](gradle.md#gradle-java-toolchains-support) instead
+>
+> **Deprecation cycle:**
+>
+> - 1.5.30: raise the deprecation level to a warning
+> - \> 1.7.0: remove the option
+
+### Remove noStdlib compiler option
+
+> **Issue**: [KT-49011](https://youtrack.jetbrains.com/issue/KT-49011)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `noStdlib` compiler option. Gradle always resolves the Kotlin standard library and adds it to the compiler classpath
+>
+> **Deprecation cycle:**
+>
+> - 1.5.30: raise the deprecation level to a warning
+> - 1.7.0: remove the option
+
+### Remove kotlin2js and kotlin-dce-plugin plugins
+
+> **Issue**: [KT-48276](https://youtrack.jetbrains.com/issue/KT-48276)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlin2js` and `kotlin-dce-plugin` plugins
+>
+> **Deprecation cycle:**
+>
+> - 1.4.0: raise the deprecation level to a warning
+> - 1.7.0: remove the plugins
+
+### Remove KotlinGradleSubplugin class
+
+> **Issue**: [KT-48831](https://youtrack.jetbrains.com/issue/KT-48831)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `KotlinGradleSubplugin` class. Use the `KotlinCompilerPluginSupportPlugin` class instead
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: raise the deprecation level to an error
+> - 1.7.0: remove the deprecated class
+
+### Changes in compile tasks
+
+> **Issue**: [KT-32805](https://youtrack.jetbrains.com/issue/KT-32805)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin compile tasks no longer inherit the Gradle `AbstractCompile` task and that's why 
+> the `sourceCompatibility` and `targetCompatibility` inputs are no longer available in Kotlin users' scripts. 
+> The `SourceTask.stableSources` input is no longer available. The `sourceFilesExtensions` input was removed. 
+> The deprecated `Gradle destinationDir: File` output was replaced with the `destinationDirectory: DirectoryProperty` output
```

Comment:
Also worth to mention `getClasspath()` and `setClasspath()` methods deprecation:
> All compile tasks are now using `libraries` input for list of compilation required libraries. `KotlinCompile` still has deprecated `setClasspath(FileCollection)` and `getClasspath(): FileCollection` methods.

Reply:
When were these methods deprecated?

Reply:
In KT-32805 :slightly_smiling_face:

I've wrote about it in my summary comment in this issue.

Reply:
@Tapchicoma isn't there the single deprecated `classpath: FileCollection` property instead?
> KotlinCompile still has deprecated setClasspath(FileCollection) and getClasspath(): FileCollection methods

Reply:
> still has deprecated
doesn't sound as they are deprecared in this release, it looks like they were deprecated some time ago

Reply:
Added in this commit https://github.com/JetBrains/kotlin-web-site/pull/2983/commits/e05244ba5e7b5e27b53e4d2d74b1d5d56f8406ed

Reply:
Still, what's about property `classpath` versus getter and setter methods? I though we provided the deprecated property for Kotlin Gradle DSL users.

Reply:
In What's new we wrote:
> The `KotlinCompile` task still has the deprecated Kotlin property `classpath`, which will be removed in future releases.

@Tapchicoma, should I write it instead of 
> The `setClasspath(FileCollection)` and `getClasspath(): FileCollection` methods are deprecated

?

Reply:
yeah, @ilya-g is right. Should be `classpath` property instead of `setClasspath` and `getClasspath`

Reply:
Fixed in the separate PR https://github.com/JetBrains/kotlin-web-site/pull/3005